### PR TITLE
Add COIG series datasets

### PR DESCRIPTION
**Pull request by yizhilll**

### CODE REVIEW

File: data_summaries/COIG-CQIA.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+    "coig-cqia": {
+        "Unique Dataset Identifier": "coig-cqia",
+        "Dataset Name": "COIG Quality is All You Need",
```
Comment (Muennighoff) [inline data_summaries/COIG-CQIA.json:4]:
```suggestion
        "Dataset Name": "COIG-CQIA",
```

File: data_summaries/COIG-Kun.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            "wudao"
+        ],
+        "Creators": [
+            "01.ai",
+            "Multimodal Art Projection (M-A-P)"
+        ],
+        "Licenses": [
+            {
+                "License": "Apache License 2.0",
+                "License URL": "https://github.com/Zheng0428/COIG-Kun/blob/master/LICENSE"
```
Comment (Muennighoff) [inline data_summaries/COIG-Kun.json:38]:
```suggestion
                "License URL": "https://www.apache.org/licenses/LICENSE-2.0"
```

I think we do it like this no @shayne-longpre ?

File: data_summaries/COIG.json [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        ],
+        "License Notes": "",
+        "License Verified By": "Yizhi Li",
+        "Dataset Filter IDs": [
+            "coig-counterfactual-correction"
+        ]
+    },
+    "coig-leetcode": {
+        "Unique Dataset Identifier": "coig-leetcode",
+        "Dataset Name": "COIG Counterfactual Correction Multi-round Chat",
```
Comment (Muennighoff) [inline data_summaries/COIG.json:201]:
```suggestion
        "Dataset Name": "COIG LeetCode",
```

File: src/preparers.py [hunk 4]
```diff
@@ -532,3 +532,40 @@ def prepare_open_orca(row):
...
+        {"from": "assistant", "text": outputs, "parent": 0},
+    ]
+
+def prepare_coig_cqia(row):
+    inputs = "".join([row['instruction'] + row['input']])
+    outputs = row['output']
+    return [
+        {"from": "user", "text": inputs, "parent": row['_source']},
+        {"from": "assistant", "text": outputs, "parent": 0},
+    ]
```
Comment (Muennighoff) [inline src/preparers.py:571]:
```suggestion
    ]

```

File: src/downloaders.py [hunk 5]
```diff
@@ -628,3 +628,139 @@ def download_gorilla(accepted_filter_ids):
...
+
+def download_coig_kun(accepted_filter_ids):
+    dset = huggingface_download('m-a-p/COIG-Kun', split='train')
+    dset = annotate_source(dset, "coig-kun")
+    return dset
+
+def download_coig_cqia(accepted_filter_ids):
+    dset = huggingface_download('m-a-p/COIG-CQIA', split='train')
+    dset = annotate_source(dset, "coig-cqia")
+    return dset
```
Comment (Muennighoff) [inline src/downloaders.py:766]:
```suggestion
    return dset

```

File: data_summaries/COIG-Kun.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{
+    "coig-kun": {
+        "Unique Dataset Identifier": "coig-kun",
+        "Dataset Name": "COIG-Kun",
+        "Paper Title": "Kun: Answer Polishment Saves Your Time for Using Intruction Backtranslation on Self-Alignment",
+        "Dataset URL": "https://github.com/Zheng0428/COIG-Kun",
+        "GitHub URL": "https://github.com/Zheng0428/COIG-Kun",
+        "Hugging Face URL": "https://huggingface.co/datasets/m-a-p/COIG-Kun",
+        "Papers with Code URL": "",
+        "ArXiv URL": "",
```
Comment (Muennighoff) [inline data_summaries/COIG-Kun.json:10]:
```suggestion
        "ArXiv URL": "https://arxiv.org/abs/2401.06477",
```

**Review by Muennighoff — COMMENTED** [review]
Looks great to me! The tests all passed right?

**Comment by yizhilll** [comment]
> Looks great to me! The tests all passed right?

Yes, all the collections have passed the testing scripts.

File: data_summaries/COIG-CQIA.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        "ArXiv URL": "",
+        "Semantic Scholar Corpus ID": "",
+        "Collection": "COIG-CQIA",
+        "Collection URL": "https://huggingface.co/datasets/m-a-p/COIG-CQIA",
+        "Languages": [
+            "Chinese"
+        ],
+        "Task Categories": [
+            "Question Answering"
+        ],
```
Comment (shayne-longpre) [inline data_summaries/COIG-CQIA.json:19]: This doesn't need to be perfect, but do you have a sense of what other tasks are covered here? Likely a few others are prominent

File: data_summaries/COIG-Kun.json (hunk 2 above)
Comment (shayne-longpre) [reply data_summaries/COIG-Kun.json:38]: I think we're fine with where it actually points to the original license, as its easier for a future reviewer/verifier to find where we made the determination.

File: data_summaries/COIG-CQIA.json [hunk 8]
```diff
@@ -0,0 +1,50 @@
+{
+    "coig-cqia": {
+        "Unique Dataset Identifier": "coig-cqia",
+        "Dataset Name": "COIG Quality is All You Need",
+        "Paper Title": "COIG-CQIA: Quality is All you need for Chinese Instruction Fine-tuning",
+        "Dataset URL": "https://github.com/paralym/COIG-CQIA",
+        "GitHub URL": "https://github.com/paralym/COIG-CQIA",
```
Comment (shayne-longpre) [inline data_summaries/COIG-CQIA.json:7]: Isn't this made up of a couple dozen sources? https://huggingface.co/datasets/m-a-p/COIG-CQIA#%E6%95%B0%E6%8D%AE%E8%AF%A6%E6%83%85

Ideally we would like to have one entry in this dictionary for each of these, so we can do a fine-grained breakdown of the Text Sources and Licenses. If you need help determining licenses, create a spreadsheet with each Dataset Text Source and we can help label them.

**Review by shayne-longpre — COMMENTED** [review]
Awesome work! It looks close, but I think we want to break these datasets down so we can fine-grained analyze their sources and licenses. That's the contribution DPI makes over HuggingFace

**Comment by yizhilll** [comment]
> Awesome work! It looks close, but I think we want to break these datasets down so we can fine-grained analyze their sources and licenses. That's the contribution DPI makes over HuggingFace

Thanks for the review! By breaking down did you specifically refer to the COIG-CQIA collection?

**Comment by shayne-longpre** [comment]
> > Awesome work! It looks close, but I think we want to break these datasets down so we can fine-grained analyze their sources and licenses. That's the contribution DPI makes over HuggingFace
> 
> Thanks for the review! By breaking down did you specifically refer to the COIG-CQIA collection?

Yes, if I understand correctly it has numerous sources that the data points came from -- is it possible to know which data points came from which source? And does COIG-Kun come from multiple sources as well? The more we can subdivide by source the more control we can exert with filters based on different source/license/other features.

**Comment by Muennighoff** [comment]
Let us know if you have thoughts @yizhilll :)
Also probably need to merge main to resolve conflicts!

**Comment by Muennighoff** [comment]
Reminder @yizhilll 😊

File: src/downloaders.py [hunk 9]
```diff
@@ -750,6 +750,156 @@ def download_gorilla(accepted_filter_ids):
...
+        accepted_filter_ids = ["coig-kun-skypile", "coig-kun-wanjuan", "coig-kun-wudao" ]
+    dset = []
+    for split in accepted_filter_ids:
+        dset_tmp = huggingface_download('m-a-p/COIG-Kun', split=split.replace('coig-kun-',''))
+        dset_tmp = annotate_source(dset_tmp, split)
+        dset = dset  + dset_tmp
+    return dset
+
+def download_coig_cqia(accepted_filter_ids):
+    if accepted_filter_ids is None:
```
Comment (shayne-longpre) [inline src/downloaders.py:890]: Should this ever be None?

File: src/downloaders.py [hunk 10]
```diff
@@ -750,6 +750,156 @@ def download_gorilla(accepted_filter_ids):
...
+                            "question": data["trans_instruction"],
+                            "answer": data["trans_output"],
+                            }]}
+                    sample['source'] = filenames_to_filter_ids[fn]
+                    all_data_list.append(sample)
+    dset = Dataset.from_list(all_data_list)
+    return dset
+
+def download_coig_kun(accepted_filter_ids):
+    if accepted_filter_ids is None:
```
Comment (shayne-longpre) [inline src/downloaders.py:880]: Should this ever be None?

**Review by shayne-longpre — COMMENTED** [review]
@Muennighoff will help prepare the csv and I will start the review of the license and source annotations, which seem to have many errors.

Left a few comments on code. And it would be great if we could run all the tests and confirm they pass without any issues or warnings. Thanks!

**Comment by shayne-longpre** [comment]
@yizhilll @Muennighoff the last part of this that we want to make sure is accurate is the annotations for the Text Sources, and Licenses. (And Task Categories might be expanded a bit.) 

Look under the "Annotation Instructions for Finetuning Datasets" section of this [doc](https://docs.google.com/document/d/1-zpjCI6MhdnlMM7euiq_jhYUhxZjZll1cD3msdqufaM/edit?usp=sharing) for specific instructions on how to annotate both of these.

Right now, for Text Sources, we are just putting the huggingface link to datasets, rather than reading up on the original sources of the text in those datasets. In cases where it pulls from many datasets like Super Natural Instructions, you can aggregate all the Text Sources [here](https://github.com/Data-Provenance-Initiative/Data-Provenance-Collection/blob/main/data_summaries/Flan%20Collection%20(Super-NaturalInstructions).json).

For Licenses, right now we are using the GitHub and HuggingFace licenses. However, the instructions from lawyers (in that google doc) is to check if any of these GitHub repos state that the license is specifically for the data (otherwise it could be for the code). And for HF dataset repos, the license is only counted if it was uploaded by the dataset creators themselves, or the Data Card explicitly justifies the chosen license in some way that can be attributed to the dataset creators. If none of these things are true, then we should put "Unspecified" as the license. If the source dataset is a huge collection like Super Natural Instructions, then just put "Various" as the license type.

So every dataset entry should have (a) the collection license that COIG attached to it, (b) the dataset specific license (or "Unspecified" if not found following conditions above), and (c) an OpenAI terms if any of it was created by that model. For datasets created from multiple datasets, you can also attach multiple for (b) if there are different ones.

**Comment by yizhilll** [comment]
@shayne-longpre @Muennighoff just updated the fields and merged conflicts. Could you folks check on the new version:?

**Review by shayne-longpre — COMMENTED** [review]
Getting very close to being done! 
- I left comments to fix certain fields (not that many left)
- Run the tests on COIG and CQIA to see that they all pass
- Verify the outputs generated through these scripts look correct

File: data_summaries/COIG-Kun.json [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+        "Languages": [
+            "Chinese"
+        ],
+        "Task Categories": [
+            "Question Answering"
+        ],
+        "Text Sources": [
+            "https://huggingface.co/datasets/Skywork/SkyPile-150B"
+        ],
+        "Model Generated": ["m-a-p/Kun-LabelModel", "01-ai/Yi-34B-Chat"],
```
[the remark this answers is not in the document]
Comment (yizhilll) [reply data_summaries/COIG-Kun.json:23]: There are 2 models used in the dataset construction process, I wrongly labelled one of them. Now this filed will be `["m-a-p/Kun-LabelModel", "01-ai/Yi-6B"]` in the COIG-Kun collection.

File: data_summaries/COIG.json [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+        "Collection": "COIG",
+        "Collection URL": "https://huggingface.co/datasets/BAAI/COIG",
+        "Languages": [
+            "Chinese"
+        ],
+        "Task Categories": [
+            "Question Answering"
+        ],
+        "Text Sources": [],
+        "Model Generated": ["DeepL"],
```
[the remark this answers is not in the document]
Comment (yizhilll) [reply data_summaries/COIG.json:21]: DeepL was used to translate the data. So I would consider it as "model generated".

File: data_summaries/COIG-CQIA.json [hunk 13]
```diff
@@ -0,0 +1,663 @@
...
+            "Sentence Completion",
+            "Translation"
+        ],
+        "Text Sources": [
+            "https://huggingface.co/datasets/YeungNLP/firefly-train-1.1M"
+        ],
+        "Model Generated": [
+            "m-a-p/Kun-LabelModel",
+            "01-ai/Yi-34B-Chat"
+        ],
```
[the remark this answers is not in the document]
Comment (yizhilll) [reply data_summaries/COIG-CQIA.json:31]: This field was wrongly filled. I removed them.

File: data_summaries/COIG-CQIA.json [hunk 14]
```diff
@@ -0,0 +1,663 @@
...
+        "Creators": [
+            "01.ai",
+            "Multimodal Art Projection (M-A-P)",
+            "Shenzhen Institute of Advanced Technology, Chinese Academy of Sciences"
+        ],
+        "Licenses": [
+            {
+                "License": "Apache License 2.0",
+                "License URL": "https://www.apache.org/licenses/LICENSE-2.0"
```
[the remark this answers is not in the document]
Comment (yizhilll) [reply data_summaries/COIG-CQIA.json:304]: Just added

File: data_summaries/COIG-CQIA.json [hunk 15]
```diff
@@ -0,0 +1,663 @@
...
+            "Chinese"
+        ],
+        "Task Categories": [
+            "Question Answering",
+            "Argumentative Question Answering",
+            "Metaphor Explanation Question Answering"
+        ],
+        "Text Sources": [
+            "https://github.com/Leymore/ruozhiba"
+        ],
```
[the remark this answers is not in the document]
Comment (shayne-longpre) [reply data_summaries/COIG-CQIA.json:372]: Yeah, either that website or the name of the website, e.g. "reddit" (or what the equivalent name is)

File: data_summaries/COIG.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (shayne-longpre) [reply data_summaries/COIG.json:21]: Perfect 👍

File: src/downloaders.py [hunk 16]
```diff
@@ -838,12 +982,10 @@ def download_bactrianx(accepted_filter_ids):
...
-
 def download_cobra_frames(accepted_filter_ids):
     mapping = {
         'normal': accepted_filter_ids[0],
     }
     dset = huggingface_download('cmu-lti/cobracorpus', data_files={'normal': 'toxigen_explanations.csv'})
     dset = annotate_source(dset['normal'], mapping['normal'])
-    return dset
-
+    return dset
```
Comment (Muennighoff) [inline src/downloaders.py:991]:
```suggestion
    return dset

```

**Comment by Muennighoff** [comment]
Could you resolve the conflicts @yizhilll ? Then I think we can merge it if you agree @shayne-longpre ?

**Comment by Muennighoff** [comment]
> > Could you resolve the conflicts @yizhilll ? Then I think we can merge it if you agree @shayne-longpre ?
> 
> Hi Niklas, I just worked out the text sources of the firefly. (just updating the code)
> 
> We are still working on the COIG-PC text sources, which might require a lot of work due to its volume. Maybe we can merge a version for now and update PC later?

Fine from my side; wdyt @shayne-longpre ?

@yizhilll also can you run the tests? (`python src/test_new_collection.py --collection "<your collection>"`)

**Comment by yizhilll** [comment]
> > > Could you resolve the conflicts @yizhilll ? Then I think we can merge it if you agree @shayne-longpre ?
> > 
> > 
> > Hi Niklas, I just worked out the text sources of the firefly. (just updating the code)
> > We are still working on the COIG-PC text sources, which might require a lot of work due to its volume. Maybe we can merge a version for now and update PC later?
> 
> Fine from my side; wdyt @shayne-longpre ?
> 
> @yizhilll also can you run the tests? (`python src/test_new_collection.py --collection "<your collection>"`)

@Muennighoff I have resolved the conflicts and the comments (except for COIG-PC). All tests passed now.

**Review by shayne-longpre — APPROVED** [review]
Awesome work @yizhilll! Thank you for following up and making all these detailed changes!

Merging it.

**Pull request merged**